### ISSUE TITLE
Lot of debug log lines after upgrading v2.27.1 to v2.28.0

### ISSUE THREAD

**Issue by dbluxo** [prometheus/prometheus#9012]
**What did you do?**

I have updated Prometheus from v2.27.1 to v2.28.0.

**What did you expect to see?**

Same log behavior as under v2.27.1

**What did you see instead? Under which circumstances?**

Debug logs are written although no debug log level is enabled:

```bash
ts=2021-06-29T09:06:41.521Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Endpoints total 9 items received"
ts=2021-06-29T09:06:41.523Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/endpoints?allowWatchBookmarks=true&resourceVersion=58644916&timeout=9m22s&timeoutSeconds=562&watch=true 200 OK in 2 milliseconds"
ts=2021-06-29T09:06:56.672Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: forcing resync"
ts=2021-06-29T09:06:56.898Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: forcing resync"
ts=2021-06-29T09:07:04.514Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Pod total 4 items received"
ts=2021-06-29T09:07:04.516Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/pods?allowWatchBookmarks=true&resourceVersion=58644406&timeout=8m28s&timeoutSeconds=508&watch=true 200 OK in 1 milliseconds"
ts=2021-06-29T09:11:43.595Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Pod total 7 items received"
ts=2021-06-29T09:11:43.597Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/pods?allowWatchBookmarks=true&resourceVersion=58645455&timeout=5m44s&timeoutSeconds=344&watch=true 200 OK in 1 milliseconds"
ts=2021-06-29T09:12:29.575Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: forcing resync"
ts=2021-06-29T09:12:29.577Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: forcing resync"
ts=2021-06-29T09:12:29.688Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: forcing resync"
ts=2021-06-29T09:13:19.261Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Service total 0 items received"
ts=2021-06-29T09:13:19.263Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/services?allowWatchBookmarks=true&resourceVersion=58627285&timeout=5m46s&timeoutSeconds=346&watch=true 200 OK in 1 milliseconds"
ts=2021-06-29T09:13:22.013Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Service total 0 items received"
ts=2021-06-29T09:13:22.015Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/services?allowWatchBookmarks=true&resourceVersion=58627285&timeout=6m55s&timeoutSeconds=415&watch=true 200 OK in 1 milliseconds"
ts=2021-06-29T09:14:30.520Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Pod total 9 items received"
ts=2021-06-29T09:14:30.522Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/pods?allowWatchBookmarks=true&resourceVersion=58645455&timeout=5m57s&timeoutSeconds=357&watch=true 200 OK in 1 milliseconds"
ts=2021-06-29T09:15:32.516Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Pod total 10 items received"
ts=2021-06-29T09:15:32.519Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/pods?allowWatchBookmarks=true&resourceVersion=58648670&timeout=7m20s&timeoutSeconds=440&watch=true 200 OK in 2 milliseconds"
ts=2021-06-29T09:16:03.523Z caller=log.go:124 component=k8s_client_runtime level=debug func=Verbose.Infof msg="pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Watch close - *v1.Endpoints total 11 items received"
ts=2021-06-29T09:16:03.525Z caller=log.go:124 component=k8s_client_runtime level=debug func=Infof msg="GET https://172.30.0.1:443/api/v1/namespaces/observability-labor/endpoints?allowWatchBookmarks=true&resourceVersion=58648895&timeout=7m41s&timeoutSeconds=461&watch=true 200 OK in 1 milliseconds"
```

**Environment**

* System information:

`Linux 4.18.0-240.22.1.el8_3.x86_64 x86_64`

* Prometheus version:

```
prometheus, version 2.28.0 (branch: HEAD, revision: ff58416a0b0224bab1f38f949f7d7c2a0f658940)
  build user:       root@32b9079a2740
  build date:       20210621-15:45:36
  go version:       go1.16.5
  platform:         linux/amd64
```

* Prometheus configuration file:
```yaml
      args:
        - '--storage.tsdb.retention.size=6GB'
        - '--config.file=/etc/config/prometheus.yml'
        - '--storage.tsdb.path=/data'
        - '--web.console.libraries=/etc/prometheus/console_libraries'
        - '--web.console.templates=/etc/prometheus/consoles'
        - '--web.enable-lifecycle'
        - '--web.external-url=https://prometheus-server-observability-labor.<redacted>'
```

```yaml
global:
  scrape_interval: 30s
  scrape_timeout: 15s
  evaluation_interval: 1m
  external_labels:
    alertmanager_url: https://alertmanager-observability-labor.<redacted>
    grafana_url: https://grafana-observability-labor.<redacted>
    prometheus_url: https://prometheus-server-observability-labor.<redacted>/alerts
alerting:
  alertmanagers:
  - authorization:
      type: Bearer
      credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
    tls_config:
      ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
      insecure_skip_verify: true
    follow_redirects: true
    scheme: http
    timeout: 10s
    api_version: v2
    relabel_configs:
    - source_labels: [__meta_kubernetes_namespace]
      separator: ;
      regex: observability-labor
      replacement: $1
      action: keep
    - source_labels: [__meta_kubernetes_pod_label_app]
      separator: ;
      regex: maas-openshift
      replacement: $1
      action: keep
    - source_labels: [__meta_kubernetes_pod_label_component]
      separator: ;
      regex: alertmanager
      replacement: $1
      action: keep
    - source_labels: [__meta_kubernetes_pod_container_port_number]
      separator: ;
      regex: null
      replacement: $1
      action: drop
    kubernetes_sd_configs:
    - role: pod
      kubeconfig_file: ""
      follow_redirects: true
      namespaces:
        names:
        - observability-labor
rule_files:
- /etc/config/rules.yml
- /etc/config/alerts.yml
scrape_configs:
- job_name: kubernetes-service-endpoints
  honor_labels: true
  honor_timestamps: true
  scrape_interval: 30s
  scrape_timeout: 15s
  metrics_path: /metrics
  scheme: http
  authorization:
    type: Bearer
    credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
  follow_redirects: true
  relabel_configs:
  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape]
    separator: ;
    regex: "true"
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
    separator: ;
    regex: (https?)
    target_label: __scheme__
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_path]
    separator: ;
    regex: (.+)
    target_label: __metrics_path__
    replacement: $1
    action: replace
  - source_labels: [__address__, __meta_kubernetes_service_annotation_prometheus_io_port]
    separator: ;
    regex: ([^:]+)(?::\d+)?;(\d+)
    target_label: __address__
    replacement: $1:$2
    action: replace
  - separator: ;
    regex: __meta_kubernetes_service_label_(.+)
    replacement: $1
    action: labelmap
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: kubernetes_namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_node_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_node
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_container_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: metric_source
    replacement: maas-prometheus
    action: replace
  kubernetes_sd_configs:
  - role: endpoints
    kubeconfig_file: ""
    follow_redirects: true
    namespaces:
      names:
      - observability-labor
- job_name: kubernetes-services
  honor_labels: true
  honor_timestamps: true
  params:
    module:
    - http_2xx
  scrape_interval: 30s
  scrape_timeout: 15s
  metrics_path: /probe
  scheme: http
  authorization:
    type: Bearer
    credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
  follow_redirects: true
  relabel_configs:
  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_probe]
    separator: ;
    regex: "true"
    replacement: $1
    action: keep
  - source_labels: [__address__]
    separator: ;
    regex: (.*)
    target_label: __param_target
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: __address__
    replacement: blackbox
    action: replace
  - source_labels: [__param_target]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - separator: ;
    regex: __meta_kubernetes_service_label_(.+)
    replacement: $1
    action: labelmap
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: kubernetes_namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_name
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: metric_source
    replacement: maas-prometheus
    action: replace
  kubernetes_sd_configs:
  - role: service
    kubeconfig_file: ""
    follow_redirects: true
    namespaces:
      names:
      - observability-labor
- job_name: kubernetes-pods
  honor_labels: true
  honor_timestamps: true
  scrape_interval: 30s
  scrape_timeout: 15s
  metrics_path: /metrics
  scheme: http
  authorization:
    type: Bearer
    credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
  follow_redirects: true
  relabel_configs:
  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
    separator: ;
    regex: "true"
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
    separator: ;
    regex: (https?)
    target_label: __scheme__
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
    separator: ;
    regex: (.+)
    target_label: __metrics_path__
    replacement: $1
    action: replace
  - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
    separator: ;
    regex: ([^:]+)(?::\d+)?;(\d+)
    target_label: __address__
    replacement: $1:$2
    action: replace
  - separator: ;
    regex: __meta_kubernetes_pod_label_(.+)
    replacement: $1
    action: labelmap
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: kubernetes_namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_container_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_port_name]
    separator: ;
    regex: prom-false
    replacement: $1
    action: drop
  - source_labels: [__meta_kubernetes_pod_container_port_name]
    separator: ;
    regex: prom-t-actuator
    replacement: $1
    action: drop
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: metric_source
    replacement: maas-prometheus
    action: replace
  kubernetes_sd_configs:
  - role: pod
    kubeconfig_file: ""
    follow_redirects: true
    namespaces:
      names:
      - observability-labor
- job_name: kubernetes-multicontainers-per-pod-with-actuator
  honor_labels: true
  honor_timestamps: true
  scrape_interval: 30s
  scrape_timeout: 15s
  metrics_path: /metrics
  scheme: http
  authorization:
    type: Bearer
    credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
  follow_redirects: true
  relabel_configs:
  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
    separator: ;
    regex: "true"
    replacement: $1
    action: keep
  - separator: ;
    regex: (.*)
    target_label: __metrics_path__
    replacement: /actuator/prometheus
    action: replace
  - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
    separator: ;
    regex: ([^:]+)(?::\d+)?;(\d+)
    target_label: __address__
    replacement: $1:$2
    action: replace
  - separator: ;
    regex: __meta_kubernetes_pod_label_(.+)
    replacement: $1
    action: labelmap
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: kubernetes_namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_name]
    separator: ;
    regex: (.*)
    target_label: kubernetes_pod_container_name
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_container_port_name]
    separator: ;
    regex: prom-t-actuator
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: metric_source
    replacement: maas-prometheus
    action: replace
  kubernetes_sd_configs:
  - role: pod
    kubeconfig_file: ""
    follow_redirects: true
    namespaces:
      names:
      - observability-labor


```

* Logs:
see above

**Comment by roidelapluie** [comment]
Thanks for you report. It might be related to our switch to go-log/kit. We should update the dependencies. Short term we could also have a replace in our go.mod.

**Comment by LeviHarrison** [comment]
I can work on this.

**Comment by roidelapluie** [comment]
@juliusv this will require a v2.28.1

**Comment by LeviHarrison** [comment]
Waiting on merge and new release of `klog-gokit`

**Comment by roidelapluie** [comment]
Fixed by #9021 released in 2.28.1

**Comment by dbluxo** [comment]
Thanks!
